src/main/java/github/scarsz/discordsrv/listeners/DiscordChatListener.java

```diff
@@ -59,9 +59,21 @@
     @Override
     public void onGuildMessageReceived(GuildMessageReceivedEvent event) {
         // if message is from null author or self do not process
-        if (event.getMember() == null || DiscordUtil.getJda() == null || event.getAuthor().equals(DiscordUtil.getJda().getSelfUser()))
+        if ((event.getMember() == null && !event.isWebhookMessage()) || DiscordUtil.getJda() == null || event.getAuthor().equals(DiscordUtil.getJda().getSelfUser()))
             return;
 
+        // block webhooks
+        if (event.isWebhookMessage()) {
+            if (DiscordSRV.config().getBoolean("DiscordChatChannelBlockWebhooks")) {
+                DiscordSRV.debug(Debug.DISCORD_TO_MINECRAFT, "Received Discord message from webhook" + event.getAuthor() + " but DiscordChatChannelBlockWebhooks is on");
+                return;
+            }
+
+           String webhookId = WebhookUtil.getWebhookUrlToUseForChannel(event.getChannel()).split("/")[5];
```

Comment:
This might cause a webhook to be created unwatendly (when using webhooks is disabled)

Reply:
Updated

Reply:
It should be checking before it even tries to figure out the webhooks, otherwise might run into a permission exception